src/js/game.ts

```diff
@@ -772,13 +779,24 @@ export class Game {
    * @param audio The audio file to play.
    * @param loop Indicates if the audio file should loop.
    */
-  playAudio(audio: string, loop = false) {
+  playAudio(audio: string, loop = false, startTime:number = 0,
+      endTime:number = undefined) {
+    let thisAudio = this.audioSources[audio];
     if (loop) {
-      this.audioSources[audio].loop = true;
+      thisAudio.loop = true;
     }
     if (!this.audioIsPlaying(audio)) {
-      this.audioSources[audio].currentTime = 0;
-      let playPromise = this.audioSources[audio].play();
+      thisAudio.currentTime = startTime;
+      let playPromise = thisAudio.play();
+
+      if (endTime !== undefined) {
+        thisAudio.addEventListener('timeupdate', function delayedPause() {
```

Comment:
Can you write this event listener code using arrow functions similar to how it's done [here](https://github.com/google/emoji-scavenger-hunt/blob/master/src/js/ui.ts#L746) Same idea with attaching the event and removing it after it is done.